osu.Game/Graphics/Carousel/Carousel.cs

```diff
@@ -569,7 +574,10 @@ protected override void Update()
             if (!selectionValid.IsValid)
             {
                 refreshAfterSelection();
+
+                // Always scroll to selection in this case (regardless of `UserScrolling` state), centering the selection.
```

Comment:
I'm only like 80% sure this bypassing the user scrolling flag is ok to do based on the usages of `selectionValid` (and `CurrentSelection.set`, by proxy), but I'll go with it for now and see how it goes I guess?

Reply:
I was hoping the xmldoc would make it obvious, but if you follow the invalidation change, this only gets reached when selection *actually changes*. And we always reset scroll in such cases.

Reply:
Narrator: It was not fine.